example/mini_core.rs

```diff
@@ -0,0 +1,130 @@
+#![feature(no_core, lang_items, rustc_attrs, intrinsics, decl_macro)]
```

Comment:
This file has a lot of extra comments that should be removed before merge. Also should get a `//!` comment saying what its purpose is, even though it's semi-obvious.

Reply:
`tests/auxiliary/` would probably be a better place for this file, since it's not really an example. And this would mirror https://github.com/rust-lang/rust/pull/130693.